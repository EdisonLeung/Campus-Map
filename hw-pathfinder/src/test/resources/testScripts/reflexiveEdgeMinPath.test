# tests pathfinding where a node has a reflective edge and multiple
# paths to another node

# Create graph
CreateGraph minPath

# Add nodes
AddNode minPath a
AddNode minPath b
AddNode minPath c

# Add edges
AddEdge minPath a b 1.0
AddEdge minPath b a 3.0
AddEdge minPath b c 3.0
AddEdge minPath a c 5.0

# test pathfinding
FindPath minPath a c

FindPath minPath b a

FindPath minPath c a