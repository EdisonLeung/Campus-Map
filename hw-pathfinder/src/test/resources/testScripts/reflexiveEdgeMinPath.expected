# tests pathfinding where a node has a reflective edge and multiple
# paths to another node

# Create graph
created graph minPath

# Add nodes
added node a to minPath
added node b to minPath
added node c to minPath

# Add edges
added edge 1.000 from a to b in minPath
added edge 3.000 from b to a in minPath
added edge 3.000 from b to c in minPath
added edge 5.000 from a to c in minPath

# test pathfinding
path from a to c:
a to b with weight 1.000
b to c with weight 3.000
total cost: 4.000

path from b to a:
b to a with weight 3.000
total cost: 3.000

path from c to a:
no path found